Data-Structure-Linked-Lists.md

```diff
@@ -0,0 +1,403 @@
+    void size()
+    {
+        int length = 0;
+        struct node *current;
+
+        for(current = head; current != NULL; current = current->next)
+        {
+            length++;
+        }
+
+        printf("Size of Linked List is %d\n", length);
+    }
+
+    // Find node with given data
+    void find(int data){
+
+        // Start from the head
+        struct node* current = head;
+
+        // If list is empty
+        if(head == NULL)
+        {
+            printf("List is empty\n");
+            return;
+        }
+
+        // Traverse through list
+        while(current->data != data){
+
+            // If it is last node
+            if(current->next == NULL){
+                printf("Not Found\n");
+                return;
+            }
+            else{
+                // Go to next node
+                current = current->next;
+            }
+        }
+
+        // If data found
+        printf("Found\n");
+    }
+
+    // Delete a node with given data
+    void delete(int data){
+
+        // Start from the first node
+        struct node* current = head;
+        struct node* previous = NULL;
+
+        // If list is empty
+        if(head == NULL){
+            printf("List is empty\n");
+            return ;
+        }
+
+        // Navigate through list
+        while(current->data != data){
+
+            // If it is last node
+            if(current->next == NULL){
+                printf("Element not found\n");
+                return ;
+            }
+            else {
+                // Store reference to current node
+                previous = current;
+                // Move to next node
+                current = current->next;
+            }
+
+        }
+
+        // Found a match, update the node
+        if(current == head) {
+            // Change head to point to next node
+            head = head->next;
+        }
+        else {
+            // Skip the current node
+            previous->next = current->next;
+        }
+
+        // Free space used by deleted node
+        free(current);
+        printf("Deleted succesfully\n");
+    }
+
+    int main() {
+        insertFirst(10); // prints Inserted successfully
+        insertFirst(20); // prints Inserted successfully
+        insertFirst(30); // prints Inserted successfully
+        insertFirst(1); // prints Inserted successfully
+        insertFirst(40); // prints Inserted successfully
+        insertFirst(56); // prints Inserted successfully
+
+        // print list
+        printList(); // prints 56 40 1 30 20 10
+
+        deleteFirst(); // prints Deleted successfully
+
+        printList(); // prints 40 1 30 20 10
+
+        find(4); // prints Not Found
+        find(1); // prints Found
+
+        delete(4); // prints Element not found
+        delete(1); // prints Deleted succesfully
+
+        printList(); // prints 40 30 20 10
+
+        return 0;
+    }
+```
+:rocket: [Run Code](https://repl.it/CVwG)
+
```

Comment:
I don't have any issues with C implementation, except that let's be consistent about it. If you can, please go back to earlier contributed articles and add C implementation. It would look odd if you implement it in one place with C / C++ but not in another.